cheat-sheet.md

```diff
@@ -233,28 +233,51 @@ Download agent logs
 Media Controller API for chat - ***New,ChatJS Required***
 
     connect.contact(function(contact){
-    contact.getAgentConnections().forEach(function(connection){
-      // chat users
-      if(connection.getMediaType() === connect.MediaType.CHAT){
-          contact.getConnection().getMediaController()
-          .then(function(controller){
-              controller.onMessage(function(response){
-                  console.log("data", response)
-              })
-
-              controller.sendMessage({message: "so and so", contentType: "text/plain"})
-                  .then(function(res, req){
-                    console.log(res.status)
-              });
-              
-              controller.onDisconnect(function(){
-                  console.log("on disconnect");
-              })
-          });
-      }
+	// WE HAVE .getAgentConnection() [Singular] and not .getAgentConnections() [Plural]
+	// We fetch agentId from .getAgentConnection() to compare with the connections that we would get from .getConnections()
+	
+	var connectionId = contact.getAgentConnection()['connectionId']
+	
+	// WE HAVE .getConnections() [Plural] and not .getConnection() [Singular]
+        //It returns an array with connection between the AGENT and CUSTOMER 
+	
+	contact.getConnections().forEach(function (connection) {
+	//console.log(connection)
+	
+	//connection.getMediaType() === chat , AND  !== connect.MediaType.CHAT
+        // connection.getMediaType() returns "chat" and not connect.MediaType.CHAT
+		
+		if (connection.getMediaType() === "chat" && connection['connectionId'] === connectionId) {
+			contact.getAgentConnection().getMediaController()
+				.then(function (controller) {
+					controller.onMessage(function (response) {
+						var data = response['data']
+						console.log(data)
+						if (data['ContentType'] === "text/plain") {
+							console.log(data['DisplayName'], ":", data['Content'])
+						}
+					})
+
+					controller.sendMessage({ message: "hey ", contentType: "text/plain" })
+						.then(function (res, req) {
+							console.log(res.httpResponse.statusCode)
+							//console.log(res.status)
+					});
+
+					// there is no such function as "onDisconnect()" in connect-streams.js 
```

Comment:
Yes, you are right. This code snippet relies on Chat functionality, meaning that ChatJS should be used as well. There are notes in Documentation.md and README.md about this. `controller.onDisconnect()` comes from ChatJS's `ChatController`.

Reply:
Can you change this to be more clear about where onDisconnect comes from?

Reply:
yes sure